cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -579,6 +580,19 @@ func (s *Service) createOrGetForwardingRule(ctx context.Context, lbname string,
 		}
 	}
 
+	// Labels on ForwardingRules must be added after resource is created
+	labels := s.scope.AdditionalLabels()
+	if !reflect.DeepEqual(labels, forwarding.Labels) {
```

Comment:
Just to confirm, this is a map right? So ordering should not be an issue while diffing correct?

Reply:
Yes it is a map. There's actually a predefined function https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/api/v1beta1/labels.go#L29-L31 that I changed to use.